public/js/reducers/sources.js

```diff
@@ -16,7 +16,9 @@ const initialState = fromJS({
 function update(state = initialState, action) {
   switch (action.type) {
     case constants.ADD_SOURCE:
-      return state.mergeIn(["sources", action.source.actor], action.source);
+      let newSource = action.source;
```

Comment:
I'd like to avoid mutating something that came in because it might be referenced elsewhere. What do you think of this?

```
let source = Object.assign({}, action.source, {
  filename: getFilenameFromUrl(newSource.url)
})
```

This also seems like something that the component could do when selecting this data, but I'm fine putting it here instead. I just realized that we're going to have to think through how we represent various things like sources and locations in a world where we will be talking to multiple servers (firefox/chrome/etc). They each will return different types of objects. Now sure how that will look yet, but we'll figure it. (For example, Chrome doesn't have an "actor" but has "scriptId")